src/settings/rageshake.ts

```diff
@@ -467,13 +467,37 @@ declare global {
  */
 export async function init(): Promise<void> {
   global.mx_rage_logger = new ConsoleLogger();
+
+  // configure loglevel based loggers:
   setLogExtension(logger, global.mx_rage_logger.log);
   // these are the child/prefixed loggers we want to capture from js-sdk
   // there doesn't seem to be an easy way to capture all children
   ["MatrixRTCSession", "MatrixRTCSessionManager"].forEach((loggerName) => {
     setLogExtension(logger.getChild(loggerName), global.mx_rage_logger.log);
   });
 
+  // intercept console logging so that we can get matrix_sdk logs:
+  // this is nasty, but no logging hooks are provided
+  (
+    ["trace", "debug", "info", "warn", "error"] as (
+      | "trace"
+      | "debug"
+      | "info"
+      | "warn"
+      | "error"
+    )[]
+  ).forEach((level) => {
+    if (!window.console[level]) return;
+    const prefix = `${level.toUpperCase()} matrix_sdk`;
+    const originalMethod = window.console[level];
+    window.console[level] = (...args): void => {
+      originalMethod(...args);
+      if (typeof args[0] === "string" && args[0].startsWith(prefix)) {
+        global.mx_rage_logger.log(LogLevel[level], "matrix_sdk", ...args);
```

Comment:
Is passing the entire `args` array intentional, or should `args[0]` (i.e. the argument with the prefix) be excluded?

Reply:
This is intended. I've pushed some comments to try and make this clearer. 🤞